lib/puppet/provider/parsedfile.rb

```diff
@@ -260,12 +268,19 @@ def self.prefetch_all_targets(resources)
 
   # Prefetch an individual target.
   def self.prefetch_target(target)
-
     begin
       target_records = retrieve(target)
     rescue Puppet::Util::FileType::FileReadError => detail
-      puts detail.backtrace if Puppet[:trace]
-      Puppet.err _("Could not prefetch %{resource} provider '%{name}' target '%{target}': %{detail}. Treating as empty") % { resource: self.resource_type.name, name: self.name, target: target, detail: detail }
+      unless @swallow_prefetch_errors
```

Comment:
Can we switch this to `if`/`else` instead of `unless`/`else`?

Reply:
Updated